galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
     def iteritems(self):
         return self.dict.iteritems()
 
+def makeCOSMOSCatalog(file_name, use_real=True, image_dir=None, dir=None, noise_dir=None,
+                      preload=False, deep_sample=False, exclude_fail=True):
+    """
+    This routine makes a catalog of galaxies based on the COSMOS sample with F814W<23.5.
+
+    Depending on the keyword arguments, particularly `use_real`, the catalog will either have
+    information about real galaxies, or parametric ones.  To use this with either type of galaxies,
+    you need to get the COSMOS datasets in the format that GalSim recognizes; see
+
+        https://github.com/GalSim-developers/GalSim/wiki/RealGalaxy-Data
```

Comment:
Three points about this:
1. This wiki page should have a direct link to the relevant files.  Making people peruse through the "Other Files" on the Great3 site is a bit unwieldy.
2. You say that they might not need the full tarball if they are going to use `use_real=False`.  But I can't figure out what file they would need to download in that case.
3. Considering that we now have the `share` directory for stashing non-code files that can be used by the code, it would be nice if we could provide an easy way for the user to download COSMOS catalog and put it there.  We don't want to do that by default, but perhaps we could distribute an executable `galsim_download_cosmos` that would do that and put it in the appropriate directory.  Then users could have `file_name=None` which would go look for the file in that directory and if it's not there, tell them about this program they can use to download it.